bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -93,25 +98,30 @@ public void dispose() {
 
   public void refresh(PJLinkDeviceConfiguration config) {
     // Do not poll if device is offline
-    if (PJLinkDeviceHandler.this.getThing().getStatusInfo().getStatus() != ThingStatus.ONLINE) {
+    if (PJLinkDeviceHandler.this.getThing().getStatus() != ThingStatus.ONLINE) {
       PJLinkDeviceHandler.this.logger.debug("Not polling device status because device is offline");
       // setup() will schedule a new refresh interval after successful reconnection, cancel this one
       this.clearRefreshInterval();
       return;
     }
 
     PJLinkDeviceHandler.this.logger.debug("Polling device status...");
-    if (config.refreshPower) {
+    if (config.refreshPower && PJLinkDeviceHandler.this.getThing().getStatus() == ThingStatus.ONLINE) {
```

Comment:
Why do you add this check ? It will always be true due to your previous test at line 101 ?
Same thing at lines 112, 117 and 120.

Reply:
The previous handleCommand could have set the device to "offline" in case errors occured.

Reply:
Sorry, I don't understand the second part of your sentence. Both the "rather" als well as the "instead of" sound the same to me. The code does the "duplicate" checks to "rather check once too much", so it does not contradict your sentence.

Also, the checks are not exactly duplicate because something might have changed since the last time we checked (because there is a handleCommand() in between).

To answer your question, nothing will break if I remove the repeated checks. I just thought it makes sense to stop doing requests as soon as we learn that the device is offline. 

If I should remove those checks, it feels like I also should not cancel the refresh interval but just continue to try to refresh, regardless of whether the device is online or not. What do you think?